### PR TITLE
Add module docstrings and reformat remaining old-format docstrings

### LINKED ISSUE #140: Fix docstrings format
This issue will track re-formatting all docstrings to [NumPy style](https://numpydoc.readthedocs.io/en/latest/format.html).

* [x] adapters
* [x] analysis
* [x] datasets
* [x] raggedarray
* [x] haversine
* [x] sphere

### PR DESCRIPTION
**Pull request by milancurcic**
Closes #140.

### CODE REVIEW

File: clouddrift/haversine.py [hunk 1]
```diff
@@ -1,3 +1,9 @@
+"""
+This module provides various Haversine functions to compute distances and
```
Comment (selipot) [inline clouddrift/haversine.py:2]: kinda makes me think that this module could be folded in the sphere module?

Comment (selipot) [reply clouddrift/haversine.py:2]: but not now

File: clouddrift/analysis.py [hunk 2]
```diff
@@ -1,3 +1,8 @@
+"""
+This module provides the common Lagrangian analysis and transformation
```
Comment (selipot) [inline clouddrift/analysis.py:2]: remove "the"

**Review by selipot — CHANGES_REQUESTED** [review]

File: clouddrift/haversine.py (hunk 1 above)
Comment (milancurcic) [reply clouddrift/haversine.py:2]: Yes, I think that would be appropriate.

File: clouddrift/analysis.py (hunk 2 above)
Comment (milancurcic) [inline clouddrift/analysis.py:2]:
```suggestion
This module provides common Lagrangian analysis and transformation
```

**Pull request merged**